tests/test_commands.py

```diff
@@ -217,3 +217,36 @@ def test_check_command_with_node_check(self, cmd):
 
         command(cmd, 'nodes')
         cmd.logger.info.assert_called_with('NODE CHECK OK')
+
+class MultilineInputTest(TestCase):
+
+    def test_multiline_sql(self):
+        cmd = CrateShell()
+        cmd._exec_and_print = MagicMock()
+        cmd.process("SELECT\n1;\nSELECT 2;")
+        cmd._exec_and_print.assert_has_calls([
+            call("SELECT\n1;"),
+            call("SELECT 2;")
+        ])
+
+    def test_multiinput_multiline_sql(self):
+        cmd = CrateShell()
+        cmd._exec_and_print = MagicMock()
+        cmd.process_iterable(["SELECT\n1;\nSELECT 2;", "SELECT 3;"])
+        cmd._exec_and_print.assert_has_calls([
+            call("SELECT\n1;"),
+            call("SELECT 2;"),
+            call("SELECT 3;"),
+
+        ])
+
+    def test_input_commands_amd_sql(self):
+        cmd = CrateShell()
+        cmd._try_exec_cmd = MagicMock()
+        cmd._exec_and_print = MagicMock()
+        cmd.process_iterable(["SELECT 1;\n\\?", "SELECT 2;"])
+        cmd._try_exec_cmd.assert_called_once_with("?")
+        cmd._exec_and_print.assert_has_calls([
+            call("SELECT 1;"),
+            call("SELECT 2;"),
+        ])
```

Comment:
At https://github.com/crate/crash/pull/350#issuecomment-748878550, @chaudum outlined an SQL statement including a JavaScript function. With GH-388, I added software tests including SQL comments.

It will be sweet if a corresponding test case could be added here, to verify all together will start to work well after bringing in a real SQL parser.

- Multiline SQL statements including newline characters.
- Multiline JavaScript functions including newline characters.
- Multiple SQL statements.
- Comments in SQL and JavaScript.

Reply:
After #388 gets merged, I will rebase and enhance the testing suite to encompass all cases.